core-httpclient-impl/src/main/java/com/optimizely/ab/OptimizelyFactory.java

```diff
@@ -248,12 +267,21 @@ public static Optimizely newDefaultInstance(String sdkKey, String fallback) {
      */
     public static Optimizely newDefaultInstance(String sdkKey, String fallback, String datafileAccessToken) {
         NotificationCenter notificationCenter = new NotificationCenter();
-
-        HttpProjectConfigManager.Builder builder = HttpProjectConfigManager.builder()
-            .withDatafile(fallback)
-            .withNotificationCenter(notificationCenter)
-            .withSdkKey(sdkKey);
-
+        HttpProjectConfigManager.Builder builder;
+        if (customHttpClient != null) {
```

Comment:
this condition should only set 
`.withOptimizelyHttpClient` like we have for `datafileAccessToken`. 
or without if condition you may set null using ternary operator that won't have any impact on `OptimizelyHttpClient`. 

`.withOptimizelyHttpClient( customHttpClient  == null ? null : OHC instance`